packages/blockchain-wallet-v4-frontend/src/components/Form/TextInputWithClipboard/index.tsx

```diff
@@ -56,9 +60,11 @@ const TextInputWithClipboard: React.FC<Props> = ({ value }) => {
     <Wrapper>
       <StyledInput disabled value={value} />
       <IconWrapper onClick={handleOnClickToCopyText}>
-        <Icon label='copy' color={showCopiedIcon ? 'green600' : 'blue600'}>
-          {showCopiedIcon ? <IconCheck /> : <IconClipboard />}
-        </Icon>
+        {showCopiedIcon ? (
+          <IconCheck color={PaletteColors['green-600']} label='copy' />
+        ) : (
+          <IconClipboard color={PaletteColors['blue-600']} label='copy' />
+        )}
```

Comment:
If the only changing thing is the Component, why not do something like
```jsx
  const IconComponent = showCopiedIcon ? IconCheck : IconClipboard;
   return (
   ...
    <IconComponent color={PaletteColors['blue-600']} label='copy' />
    ...
  )
```

Reply:
Good point, but in this case since the color is also different it won't really save many keystrokes to do it the suggested way.